language/vm/vm-runtime/src/interpreter.rs

```diff
@@ -811,8 +837,20 @@ impl<'txn> Interpreter<'txn> {
             StructDef,
         ) -> VMResult<AbstractMemorySize<GasCarrier>>,
     {
-        let ap = Self::make_access_path(module, idx, address);
-        let struct_def = runtime.resolve_struct_def(module, idx, context)?;
+        let module = frame.module();
```

Comment:
this is where the substitution of the instantiation in the instructions (the signature in the global* bytecodes) is happening in order to generate a proper instantiation

Reply:
> this is where the substitution of the instantiation in the instructions (the signature in the global* bytecodes) is happening in order to generate a proper instantiation

actually it looks good and thanks to @vgao1996 to cover my insanity in writing the test... :)